IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m

```diff
@@ -98,13 +98,52 @@ - (BOOL)updateAuthorityURL:(NSURL *)cacheAuthorityURL
                                        context:context error:error];
 }
 
-- (BOOL)clearForHomeAccountId:(NSString *)homeAccountId
-                     clientId:(NSString *)clientId
-                      context:(id<MSIDRequestContext>)context
-                        error:(NSError **)error
+- (MSIDAccountMetadataState)signInStateForHomeAccountId:(NSString *)homeAccountId
+                                               clientId:(NSString *)clientId
+                                                context:(id<MSIDRequestContext>)context
+                                                  error:(NSError **)error
 {
+    if ([NSString msidIsStringNilOrBlank:homeAccountId]
+        || [NSString msidIsStringNilOrBlank:clientId])
+    {
+        if (error) *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidInternalParameter, @"Both homeAccountId and clientId are needed to query signed out state!", nil, nil, nil, nil, nil);
+        return MSIDAccountMetadataStateUnknown;
+    }
+    
     MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWitHomeAccountId:homeAccountId clientId:clientId];
-    return [_metadataCache removeAccountMetadataForKey:key context:context error:error];
+    MSIDAccountMetadataCacheItem *acountMetadata = [_metadataCache accountMetadataWithKey:key context:context error:error];
+    if (!acountMetadata) { return MSIDAccountMetadataStateUnknown; }
+    
+    return acountMetadata.signInState;
+}
+
+- (BOOL)markSignedOutStateForHomeAccountId:(NSString *)homeAccountId
```

Comment:
should this perhaps be more generic to update state of the account?

Reply:
addressed :)